x/evm/keeper/grpc_query_test.go

```diff
@@ -447,38 +467,518 @@ func (s *KeeperSuite) TestQueryParams() {
+				coins := sdk.Coins{sdk.NewInt64Coin(evm.DefaultEVMDenom, 1000)}
+				err := chain.BankKeeper.MintCoins(deps.Ctx, evm.ModuleName, coins)
+				s.NoError(err)
+				err = chain.BankKeeper.SendCoinsFromModuleToAccount(
+					deps.Ctx, evm.ModuleName, ethAddr.Bytes(), coins)
+				s.Require().NoError(err)
+			},
+			scenario: func(deps *evmtest.TestDeps) (req In, wantResp Out) {
+				recipient := evmtest.NewEthAccInfo().EthAddr
+				amountToSend := hexutil.Big(*big.NewInt(10))
+				gasLimitArg := hexutil.Uint64(100000)
+
+				jsonTxArgs, err := json.Marshal(&evm.JsonTxArgs{
+					From:  &deps.Sender.EthAddr,
+					To:    &recipient,
+					Value: &amountToSend,
+					Gas:   &gasLimitArg,
+				})
+				s.Require().NoError(err)
+				req = &evm.EthCallRequest{
+					Args:   jsonTxArgs,
+					GasCap: gethparams.TxGas,
+				}
+				wantResp = &evm.EstimateGasResponse{
+					Gas: gethparams.TxGas,
+				}
+				return req, wantResp
+			},
+			wantErr: "",
+		},
+		{
+			name: "sad: insufficient balance for transfer",
+			scenario: func(deps *evmtest.TestDeps) (req In, wantResp Out) {
+				recipient := evmtest.NewEthAccInfo().EthAddr
+				amountToSend := hexutil.Big(*big.NewInt(10))
+
+				jsonTxArgs, err := json.Marshal(&evm.JsonTxArgs{
+					From:  &deps.Sender.EthAddr,
+					To:    &recipient,
+					Value: &amountToSend,
+				})
+				s.Require().NoError(err)
+				req = &evm.EthCallRequest{
+					Args:   jsonTxArgs,
+					GasCap: gethparams.TxGas,
+				}
+				wantResp = nil
+				return req, wantResp
+			},
+			wantErr: "insufficient balance for transfer",
+		},
+	}
+
+	for _, tc := range testCases {
+		s.Run(tc.name, func() {
+			deps := evmtest.NewTestDeps()
+			if tc.setup != nil {
+				tc.setup(&deps)
+			}
+			req, wantResp := tc.scenario(&deps)
+			goCtx := sdk.WrapSDKContext(deps.Ctx)
+			gotResp, err := deps.K.EstimateGas(goCtx, req)
+			if tc.wantErr != "" {
+				s.Require().ErrorContains(err, tc.wantErr)
+				return
+			}
+			s.Assert().NoError(err)
+			s.EqualValues(wantResp, gotResp)
+		})
+	}
+}
+
+func (s *KeeperSuite) TestTestTraceTx() {
+	type In = *evm.QueryTraceTxRequest
+	type Out = string
+
+	testCases := []TestCase[In, Out]{
+		{
+			name: "sad: nil query",
+			scenario: func(deps *evmtest.TestDeps) (req In, wantResp Out) {
+				return nil, ""
+			},
+			wantErr: "InvalidArgument",
+		},
+		{
+			name: "happy: simple nibi transfer tx",
+			scenario: func(deps *evmtest.TestDeps) (req In, wantResp Out) {
+				txMsg := s.ExecuteNibiTransfer(deps)
+				req = &evm.QueryTraceTxRequest{
+					Msg: txMsg,
+				}
+				wantResp = TraceNibiTransfer()
+				return req, wantResp
+			},
+			wantErr: "",
+		},
+		{
+			"happy: trace erc-20 transfer tx",
+			nil,
+			func(deps *evmtest.TestDeps) (req In, wantResp Out) {
+				txMsg, predecessors := s.ExecuteERC20Transfer(deps)
+
+				req = &evm.QueryTraceTxRequest{
+					Msg:          txMsg,
+					Predecessors: predecessors,
+				}
+				wantResp = TraceERC20Transfer()
+				return req, wantResp
+			},
```

Comment:
I would write the `ExecuteERC20Transfer` function as taking arguments `(deps, s)` rather than including it as a function on the suite. That will make it easier to pull out the testutil to the `evmtest` import if we want to leverage it in other places (like `nibiru/eth` or `nibiru/app`) 

Perhaps the `ExecuteNibiTransfer` fits this criterion as well.